merino/providers/__init__.py

```diff
@@ -38,6 +40,10 @@ async def init_providers() -> None:
     # register providers
     for provider_type, setting in settings.providers.items():
         match provider_type:
+            case ProviderType.ACCUWEATHER:
+                providers["accuweather"] = AccuWeatherProvider(
+                    enabled_by_default=setting.enabled_by_default,
```

Comment:
Nit: I know that `name` takes a default value "accuweather", but let's pass it explicitly just in case we change the configuration in the future.
```suggestion
                    name=provider_type,
                    enabled_by_default=setting.enabled_by_default,
```